sdk/communication/azure-communication-jobrouter/azure/communication/jobrouter/_router_client.py

```diff
@@ -0,0 +1,1219 @@
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/router_worker_crud_ops.py
+                :start-after: [START get_worker]
+                :end-before: [END get_worker]
+                :language: python
+                :dedent: 8
+                :caption: Use a RouterClient to get a worker
+        """
+        if not worker_id:
+            raise ValueError("identifier cannot be None.")
+
+        return self._client.job_router.get_worker(
+            worker_id = worker_id,
+            **kwargs
+        )
+
+    @distributed_trace
+    def list_workers(
+            self,
+            *,
+            status: Optional[Union[str, WorkerStateSelector]] = WorkerStateSelector.ALL,
+            channel_id: Optional[str] = None,
+            queue_id: Optional[str] = None,
+            has_capacity: Optional[bool] = None,
+            results_per_page: Optional[int] = None,
+            **kwargs: Any
+    ) -> ItemPaged[RouterWorkerItem]:
+        """Retrieves existing workers.
+
+        :keyword status: If specified, select workers by worker status. Default value is "all".
+          Accepted value(s): active, draining, inactive, all
+        :paramtype status: Optional[Union[str, ~azure.communication.jobrouter.WorkerStateSelector]]
+
+        :keyword channel_id: If specified, select workers who have a channel configuration
+           with this channel. Default value is None.
+        :paramtype channel_id: Optional[str]
+
+        :keyword queue_id: If specified, select workers who are assigned to this queue.
+           Default value is None.
+        :paramtype queue_id: Optional[str]
+
+        :keyword has_capacity: If set to true, select only workers who have capacity for the
+           channel specified by ``channelId`` or for any channel
+        :paramtype has_capacity: Optional[bool]
+
+        :keyword Optional[int] results_per_page: The maximum number of results to be returned per page.
+
+        :return: An iterator like instance of RouterWorkerItem
+        :rtype: ~azure.core.paging.ItemPaged[~azure.communication.jobrouter.RouterWorkerItem]
+        :raises: ~azure.core.exceptions.HttpResponseError, ValueError
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/router_worker_crud_ops.py
+                :start-after: [START list_workers]
+                :end-before: [END list_workers]
+                :language: python
+                :dedent: 8
+                :caption: Use a RouterClient to retrieve workers
+        """
+
+        params = {}
+        if results_per_page is not None:
+            params['maxpagesize'] = _SERIALIZER.query("maxpagesize", results_per_page, 'int')
+
+        if not isinstance(status, WorkerStateSelector):
+            try:
+                status = WorkerStateSelector.__getattr__(status)
+            except Exception:
+                raise ValueError(f"status: {status} is not acceptable")
```

Comment:
Confused as to what this patch is trying to do?

Reply:
This method takes in `status` of type `Union[str, WorkerStateSelector]`. I added the check to validate that when a `str` value is passed it is indeed one of the enums that is supported in the `WorkerStateSelector`

Reply:
I have removed this